src/globals/tabnineExtensionProperties.ts

```diff
@@ -79,22 +72,29 @@ function getContext(): TabNineExtensionProperties {
     .includes("insider");
 
   return {
+    get extension(): vscode.Extension<unknown> {
+      // this file is eagerly loaded before the activation event occurred.
+      // The pre-computed values can be incorrect in the case where multiple Tabnines are installed (i.e. onprem is installed along with the GA version)
+      // this is solved by calling the getter of the extension context which is set after the activation event occurred (with the correct context values)
+      return getTabnineExtensionContext().extension;
+    },
     get extensionPath(): string | undefined {
-      return extension?.extensionPath;
+      return this.extension.extensionPath;
```

Comment:
The usage of `this` could lead to an undefined behavior (ie any of these functions are used as callbacks ).
move the `get extension` to a function and just call `getExtension` instead of `this.extension`